UE-536: Corrected mobile layout and some spacing issues

jenbreese opened the pull request:
# READY FOR REVIEW

# Summary

1. I moved the paragraphs into the same area as the eligibility/prerequisites and body. - Layout change.
2. I also fixed the summary `max-width` to 924px like in the figma.
3. The right column is at 300px now.
4. Fixed the font size of an opportunity card on a opp node. 

# Review By (Date)
- 6/19

# Criticality
- High

# Urgency
- High

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Navigate to...
5. Verify...

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?  YES

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- UE-536](https://stanfordits.atlassian.net/browse/UE-536)
- https://github.com/SU-SWS/stanford_profile_helper/pull/409 - Profile helper
- https://github.com/SU-SWS/stanford_profile/pull/956 - Stanford profile
- https://github.com/SU-SWS/vpue_undergrad_subtheme/pull/76 - VPUE subtheme - not dependent on PRs above

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)
-

jenbreese committed:
UE-536: Corrected mobile layout and some spacing issues

jenbreese committed:
UE-536: adjusting layout arrangement

jenbreese committed:
removed icon class as not working exactly right.

jenbreese committed:
fixup

pookmish commented on modules/stanford_profile_styles/lib/scss/component/node/stanford_opportunity.scss:
```diff
@@ -74,6 +77,23 @@
   }
 
+  // Right sidebar
+  .main-region {
```
`.main-region` is used on every layout template, 1 column, 2 column, and 3 column. This selector does not only refer to the right sidebar. Please revise

jenbreese replied:
I restructured the css to use the wrapper classes and nested selectors accordingly.

pookmish commented on modules/stanford_profile_styles/lib/scss/component/node/stanford_opportunity.scss:
```diff
@@ -122,6 +142,12 @@
     }
   }
 
+  .su-card {
+    .su-opp-image {
```
do you need the additional selector?

jenbreese replied:
No I fixed the margin-bottom so I don't need it at all.

pookmish commented on modules/stanford_profile_styles/lib/scss/component/node/stanford_opportunity.scss:
```diff
@@ -321,42 +349,44 @@
-          color: $su-color-digital-red;
-          text-decoration: underline;
-        }
+        &.su-link--action {
```
is this a possibility? `<h3>` with `su-link--action` class?

jenbreese replied:
On a teaser for an Opportunity ct, I couldn't change the heading. I got an `h2` for the teaser headline when rendered. I got an `h3` inside the opportunity card. 

On a list of Opportunities, I also couldn't change the heading type. I got `h2` for the list headline and an `h3` inside the opportunity card.

On a card paragraph, I could select different heading types.

pookmish replied:
Ok, i see. So this needs to be changed. `su-link--action` is intended for use on an `<a>` element, not a heading element. You are adding classes via the manage display area that aren't intended for that particular element. For now I will let this go, but you need to choose better class names that aren't mixed with other element tags.

pookmish reviewed (changes requested):
Not related to the changes here, but you need to replace the `.opportunity` top selector. If we ever change the label of the content type, this will break all styles.

jenbreese committed:
moved around the css and added more comments.

jenbreese commented:
hi @pookmish I moved around a lot of css. I took out the `.opportunity` and replaced it with the `node--layout-full`. I'm not sure if that is better. ¯\_(ツ)_/¯  I reviewed and removed extra code and put it in the wrapper classes.

jenbreese committed:
fixup

pookmish commented on modules/stanford_profile_styles/lib/scss/component/node/stanford_opportunity.scss:
```diff
@@ -2,7 +2,7 @@
 @import 'stanford_page';
 
-.opportunity {
+.node--layout-full {
```
Now this will apply to all content types. This needs to be specific to the opportunities but it should reference the machine name of the content type. Something like `node--stanford-opportunity` or `stanford-opportunity`. You might need to add that class yourself. Please try to find a solution and if you get stuck, I can help.

jenbreese replied:
@pookmish How about this recent fixup to add the class?

pookmish replied:
Now none of the styles are applied. `node--stanford-opportunity` does not exist on the page at the moment.

jenbreese replied:
@pookmish Oh you need to update the stanford_profile one too.

jenbreese committed:
fixup
